src/s2s/codec/codec.cpp

```
@@ -0,0 +1,219 @@
+    // Repack codes (n_frames, Q) row-major -> q-major (BT per q contiguous).
+    std::vector<int32_t> codes_qbt(static_cast<size_t>(Q) * BT);
+    for (int q = 0; q < Q; q++)
+        for (int t = 0; t < BT; t++)
+            codes_qbt[static_cast<size_t>(q) * BT + t] = codes[static_cast<size_t>(t) * Q + q];
+
+    int out_trim = 0;
+    auto env_it = envelope_cache_.find(n_frames);
+    if (env_it == envelope_cache_.end()) {
+        env_it =
+            envelope_cache_.emplace(n_frames, compute_envelope_recip(n_frames, &out_trim)).first;
+    } else {
```

Clang (14.0.6)

[note] else branch starts here

(clang)